common/lib/xmodule/xmodule/video_module/transcripts_utils.py

```diff
@@ -531,6 +529,9 @@ def asset_location(location, filename):
         """
         Return asset location. `location` is module location.
         """
+        # If user transcript filename is empty, raise `TranscriptException` to avoid `InvalidKeyError`.
+        if not filename:
+            raise TranscriptException("Transcript not uploaded yet")
```

Comment:
After @marcotuts 's suggestion to save the language even if the associated transcript file is not uploaded, this does not result into the error on studio/lms.

However, as a precaution, I have added this so that a correct exception would be generated which may/may-not be handled (depends upon the caller of `asset_location` method). This is just a way for us to be safe from `InvalidKeyError` when no asset is present.